{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Table of Contents\n",
    " <p><div class=\"lev1\"><a href=\"#Task-1.-Compiling-Ebola-Data\"><span class=\"toc-item-num\">Task 1.&nbsp;&nbsp;</span>Compiling Ebola Data</a></div>\n",
    " <div class=\"lev1\"><a href=\"#Task-2.-RNA-Sequences\"><span class=\"toc-item-num\">Task 2.&nbsp;&nbsp;</span>RNA Sequences</a></div>\n",
    " <div class=\"lev1\"><a href=\"#Task-3.-Class-War-in-Titanic\"><span class=\"toc-item-num\">Task 3.&nbsp;&nbsp;</span>Class War in Titanic</a></div></p>"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "import matplotlib.pyplot as plt\n",
    "import glob \n",
    "#import seaborn as sbn\n",
    "%matplotlib inline\n",
    "DATA = './Data/ebola/guinea_data' # Use the data folder provided in Tutorial 02 - Intro to Pandas."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Task 1. Compiling Ebola Data\n",
    "\n",
    "The `DATA_FOLDER/ebola` folder contains summarized reports of Ebola cases from three countries (Guinea, Liberia and Sierra Leone) during the recent outbreak of the disease in West Africa. For each country, there are daily reports that contain various information about the outbreak in several cities in each country.\n",
    "\n",
    "Use pandas to import these data files into a single `Dataframe`.\n",
    "Using this `DataFrame`, calculate for *each country*, the *daily average per month* of *new cases* and *deaths*.\n",
    "Make sure you handle all the different expressions for *new cases* and *deaths* that are used in the reports."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                                                Beyla Boffa Conakry Coyah  \\\n",
      "Description                                                                 \n",
      "New cases of suspects                             NaN     0       5   NaN   \n",
      "New cases of probables                            NaN     0       0   NaN   \n",
      "New cases of confirmed                            NaN     0       1   NaN   \n",
      "Total new cases registered so far                 NaN     0       6   NaN   \n",
      "Total cases of suspects                           NaN     0       9   NaN   \n",
      "Total cases of probables                          NaN     5       8   NaN   \n",
      "Total cases of confirmed                          NaN    18      78   NaN   \n",
      "Cumulative (confirmed + probable + suspects)      NaN    23      95   NaN   \n",
      "New deaths registered today                       NaN     0       0   NaN   \n",
      "New deaths registered today (confirmed)           NaN     0       0   NaN   \n",
      "New deaths registered today (probables)           NaN     0       0   NaN   \n",
      "New deaths registered today (suspects)            NaN     0       0   NaN   \n",
      "Total deaths of suspects                          NaN     0       2   NaN   \n",
      "Total deaths of probables                         NaN     5       8   NaN   \n",
      "Total deaths of confirmed                         NaN    11      32   NaN   \n",
      "Total deaths (confirmed + probables + suspects)   NaN    16      42   NaN   \n",
      "Total PEC center today                            NaN     0      21   NaN   \n",
      "Total PEC center today (confirmed)                NaN     0      10   NaN   \n",
      "Total PEC center today (probables)                NaN     0       0   NaN   \n",
      "Total PEC center today (suspects)                 NaN     0      11   NaN   \n",
      "Total of deaths in confirmed cases in CTE         NaN     6      20   NaN   \n",
      "Total of cured in confirmed cases in CTE          NaN     7      34   NaN   \n",
      "Number of male confirmed cases                    NaN    10      48   NaN   \n",
      "Number of female confirmed cases                  NaN     8      30   NaN   \n",
      "Number of male probables cases                    NaN     3       4   NaN   \n",
      "Number of female probables cases                  NaN     2       4   NaN   \n",
      "Number of male suspects cases                     NaN     0       5   NaN   \n",
      "Number of female suspects cases                   NaN     0       4   NaN   \n",
      "Total number of male cases                        NaN    13      57   NaN   \n",
      "Total number of female cases                      NaN    10      38   NaN   \n",
      "...                                               ...   ...     ...   ...   \n",
      "New cases of confirmed                            NaN   NaN       2   NaN   \n",
      "Total new cases registered so far                   0     0      17     0   \n",
      "Total cases of suspects                             0     0      19     0   \n",
      "Total cases of probables                            0     5      10     0   \n",
      "Total cases of confirmed                            2    19     170    12   \n",
      "Cumulative (confirmed + probable + suspects)        2    24     199    12   \n",
      "Total suspected non-class cases                     0    20     355     6   \n",
      "New deaths registered                             NaN   NaN     NaN   NaN   \n",
      "Total deaths of suspects                            0     0       0     0   \n",
      "Total deaths of probables                           0     5      10     0   \n",
      "Total deaths of confirmed                           1    11      68     5   \n",
      "Total deaths (confirmed + probables + suspects)     0    16      78     5   \n",
      "Fatality rate for confirmed and probables        100%   67%     39%   42%   \n",
      "New cases of confirmed among health workers       NaN   NaN       2   NaN   \n",
      "Total case of confirmed among health workers        0     0      40     1   \n",
      "New deaths registered among health workers          0     0       0     0   \n",
      "Total deaths registered among health workers        0     0      18     0   \n",
      "New admits to CTE so far                          NaN   NaN      15   NaN   \n",
      "Total number of hospitalized cases in CTE           0     0      48     0   \n",
      "Total number of exits from CTE                      0     0       9     0   \n",
      "Total number of admissions to CTE                   0     0     571     0   \n",
      "New contacts registered so far                    NaN   NaN     NaN   NaN   \n",
      "Total contacts registered from start date           2   397    3319   116   \n",
      "Number of contacts to follow today                929     0     825    98   \n",
      "Number of contacts followed today                  29     0     778    98   \n",
      "Number of contacts out of the track 21 days         0   397    2494    17   \n",
      "Number of contacts lost to follow up                0     0       0     1   \n",
      "Number of samples collected today                 NaN   NaN      16   NaN   \n",
      "Number of samples under test                      NaN   NaN       0   NaN   \n",
      "Total samples tested                              NaN   NaN    1171   NaN   \n",
      "\n",
      "                                                Dabola Dalaba        Date  \\\n",
      "Description                                                                 \n",
      "New cases of suspects                                0    NaN  2014-08-04   \n",
      "New cases of probables                               0    NaN  2014-08-04   \n",
      "New cases of confirmed                               0    NaN  2014-08-04   \n",
      "Total new cases registered so far                    0    NaN  2014-08-04   \n",
      "Total cases of suspects                              0    NaN  2014-08-04   \n",
      "Total cases of probables                             3    NaN  2014-08-04   \n",
      "Total cases of confirmed                             1    NaN  2014-08-04   \n",
      "Cumulative (confirmed + probable + suspects)         4    NaN  2014-08-04   \n",
      "New deaths registered today                          0    NaN  2014-08-04   \n",
      "New deaths registered today (confirmed)              0    NaN  2014-08-04   \n",
      "New deaths registered today (probables)              0    NaN  2014-08-04   \n",
      "New deaths registered today (suspects)               0    NaN  2014-08-04   \n",
      "Total deaths of suspects                             0    NaN  2014-08-04   \n",
      "Total deaths of probables                            3    NaN  2014-08-04   \n",
      "Total deaths of confirmed                            1    NaN  2014-08-04   \n",
      "Total deaths (confirmed + probables + suspects)      4    NaN  2014-08-04   \n",
      "Total PEC center today                               0    NaN  2014-08-04   \n",
      "Total PEC center today (confirmed)                   0    NaN  2014-08-04   \n",
      "Total PEC center today (probables)                   0    NaN  2014-08-04   \n",
      "Total PEC center today (suspects)                    0    NaN  2014-08-04   \n",
      "Total of deaths in confirmed cases in CTE            0    NaN  2014-08-04   \n",
      "Total of cured in confirmed cases in CTE             0    NaN  2014-08-04   \n",
      "Number of male confirmed cases                       1    NaN  2014-08-04   \n",
      "Number of female confirmed cases                     0    NaN  2014-08-04   \n",
      "Number of male probables cases                       1    NaN  2014-08-04   \n",
      "Number of female probables cases                     2    NaN  2014-08-04   \n",
      "Number of male suspects cases                        0    NaN  2014-08-04   \n",
      "Number of female suspects cases                      0    NaN  2014-08-04   \n",
      "Total number of male cases                           2    NaN  2014-08-04   \n",
      "Total number of female cases                         2    NaN  2014-08-04   \n",
      "...                                                ...    ...         ...   \n",
      "New cases of confirmed                             NaN    NaN  2014-10-01   \n",
      "Total new cases registered so far                    0      0  2014-10-01   \n",
      "Total cases of suspects                              0      0  2014-10-01   \n",
      "Total cases of probables                             3      0  2014-10-01   \n",
      "Total cases of confirmed                             1      6  2014-10-01   \n",
      "Cumulative (confirmed + probable + suspects)         4      6  2014-10-01   \n",
      "Total suspected non-class cases                      3      1  2014-10-01   \n",
      "New deaths registered                              NaN    NaN  2014-10-01   \n",
      "Total deaths of suspects                             0      0  2014-10-01   \n",
      "Total deaths of probables                            3      0  2014-10-01   \n",
      "Total deaths of confirmed                            1      0  2014-10-01   \n",
      "Total deaths (confirmed + probables + suspects)      4      0  2014-10-01   \n",
      "Fatality rate for confirmed and probables         100%     0%  2014-10-01   \n",
      "New cases of confirmed among health workers        NaN    NaN  2014-10-01   \n",
      "Total case of confirmed among health workers         0      0  2014-10-01   \n",
      "New deaths registered among health workers           0      0  2014-10-01   \n",
      "Total deaths registered among health workers         0      0  2014-10-01   \n",
      "New admits to CTE so far                           NaN    NaN  2014-10-01   \n",
      "Total number of hospitalized cases in CTE            0      0  2014-10-01   \n",
      "Total number of exits from CTE                       0      0  2014-10-01   \n",
      "Total number of admissions to CTE                    0      0  2014-10-01   \n",
      "New contacts registered so far                     NaN    NaN  2014-10-01   \n",
      "Total contacts registered from start date           69     34  2014-10-01   \n",
      "Number of contacts to follow today                   0     34  2014-10-01   \n",
      "Number of contacts followed today                    0     20  2014-10-01   \n",
      "Number of contacts out of the track 21 days         69      0  2014-10-01   \n",
      "Number of contacts lost to follow up                 0      0  2014-10-01   \n",
      "Number of samples collected today                  NaN    NaN  2014-10-01   \n",
      "Number of samples under test                       NaN    NaN  2014-10-01   \n",
      "Total samples tested                               NaN    NaN  2014-10-01   \n",
      "\n",
      "                                                Dinguiraye Dubreka Forecariah  \\\n",
      "Description                                                                     \n",
      "New cases of suspects                                    0       0        NaN   \n",
      "New cases of probables                                   0       0        NaN   \n",
      "New cases of confirmed                                   0       0        NaN   \n",
      "Total new cases registered so far                        0       0        NaN   \n",
      "Total cases of suspects                                  0       0        NaN   \n",
      "Total cases of probables                                 1       0        NaN   \n",
      "Total cases of confirmed                                 0       0        NaN   \n",
      "Cumulative (confirmed + probable + suspects)             1       0        NaN   \n",
      "New deaths registered today                              0       0        NaN   \n",
      "New deaths registered today (confirmed)                  0       0        NaN   \n",
      "New deaths registered today (probables)                  0       0        NaN   \n",
      "New deaths registered today (suspects)                   0       0        NaN   \n",
      "Total deaths of suspects                                 0       0        NaN   \n",
      "Total deaths of probables                                1       0        NaN   \n",
      "Total deaths of confirmed                                0       0        NaN   \n",
      "Total deaths (confirmed + probables + suspects)          1       0        NaN   \n",
      "Total PEC center today                                   0       0        NaN   \n",
      "Total PEC center today (confirmed)                       0       0        NaN   \n",
      "Total PEC center today (probables)                       0       0        NaN   \n",
      "Total PEC center today (suspects)                        0       0        NaN   \n",
      "Total of deaths in confirmed cases in CTE                0       0        NaN   \n",
      "Total of cured in confirmed cases in CTE                 0       0        NaN   \n",
      "Number of male confirmed cases                           0       0        NaN   \n",
      "Number of female confirmed cases                         0       0        NaN   \n",
      "Number of male probables cases                           0       0        NaN   \n",
      "Number of female probables cases                         1       0        NaN   \n",
      "Number of male suspects cases                            0       0        NaN   \n",
      "Number of female suspects cases                          0       0        NaN   \n",
      "Total number of male cases                               0       0        NaN   \n",
      "Total number of female cases                             1       0        NaN   \n",
      "...                                                    ...     ...        ...   \n",
      "New cases of confirmed                                 NaN     NaN        NaN   \n",
      "Total new cases registered so far                        0       0          0   \n",
      "Total cases of suspects                                  0       0          3   \n",
      "Total cases of probables                                 1       0          2   \n",
      "Total cases of confirmed                                 0      21         11   \n",
      "Cumulative (confirmed + probable + suspects)             1      21         16   \n",
      "Total suspected non-class cases                          0      20         13   \n",
      "New deaths registered                                  NaN     NaN        NaN   \n",
      "Total deaths of suspects                                 0       0          0   \n",
      "Total deaths of probables                                1       0          2   \n",
      "Total deaths of confirmed                                0       7          7   \n",
      "Total deaths (confirmed + probables + suspects)          1       7          9   \n",
      "Fatality rate for confirmed and probables             100%     33%        56%   \n",
      "New cases of confirmed among health workers            NaN     NaN        NaN   \n",
      "Total case of confirmed among health workers             0       3          0   \n",
      "New deaths registered among health workers               0       0          0   \n",
      "Total deaths registered among health workers             0       1          0   \n",
      "New admits to CTE so far                               NaN     NaN        NaN   \n",
      "Total number of hospitalized cases in CTE                0       0          0   \n",
      "Total number of exits from CTE                           0       0          0   \n",
      "Total number of admissions to CTE                        0       0          0   \n",
      "New contacts registered so far                         NaN     NaN        NaN   \n",
      "Total contacts registered from start date                0     341        212   \n",
      "Number of contacts to follow today                       0      49        151   \n",
      "Number of contacts followed today                        0      49        146   \n",
      "Number of contacts out of the track 21 days              0     292         61   \n",
      "Number of contacts lost to follow up                     0       0          0   \n",
      "Number of samples collected today                      NaN     NaN        NaN   \n",
      "Number of samples under test                           NaN     NaN        NaN   \n",
      "Total samples tested                                   NaN     NaN        NaN   \n",
      "\n",
      "                                                 ...  Kouroussa  Lola Macenta  \\\n",
      "Description                                      ...                            \n",
      "New cases of suspects                            ...          0   NaN       0   \n",
      "New cases of probables                           ...          0   NaN       0   \n",
      "New cases of confirmed                           ...          0   NaN       0   \n",
      "Total new cases registered so far                ...          0   NaN       0   \n",
      "Total cases of suspects                          ...          0   NaN       0   \n",
      "Total cases of probables                         ...          2   NaN      11   \n",
      "Total cases of confirmed                         ...          2   NaN      28   \n",
      "Cumulative (confirmed + probable + suspects)     ...          4   NaN      39   \n",
      "New deaths registered today                      ...          0   NaN       0   \n",
      "New deaths registered today (confirmed)          ...          0   NaN       0   \n",
      "New deaths registered today (probables)          ...          0   NaN       0   \n",
      "New deaths registered today (suspects)           ...          0   NaN       0   \n",
      "Total deaths of suspects                         ...          0   NaN       0   \n",
      "Total deaths of probables                        ...          2   NaN      11   \n",
      "Total deaths of confirmed                        ...          1   NaN      18   \n",
      "Total deaths (confirmed + probables + suspects)  ...          3   NaN      29   \n",
      "Total PEC center today                           ...          0   NaN       0   \n",
      "Total PEC center today (confirmed)               ...          0   NaN       0   \n",
      "Total PEC center today (probables)               ...          0   NaN       0   \n",
      "Total PEC center today (suspects)                ...          0   NaN       0   \n",
      "Total of deaths in confirmed cases in CTE        ...          0   NaN      10   \n",
      "Total of cured in confirmed cases in CTE         ...          0   NaN       8   \n",
      "Number of male confirmed cases                   ...          1   NaN      14   \n",
      "Number of female confirmed cases                 ...          1   NaN      14   \n",
      "Number of male probables cases                   ...          2   NaN       5   \n",
      "Number of female probables cases                 ...          0   NaN       6   \n",
      "Number of male suspects cases                    ...          0   NaN       0   \n",
      "Number of female suspects cases                  ...          0   NaN       0   \n",
      "Total number of male cases                       ...          3   NaN      19   \n",
      "Total number of female cases                     ...          1   NaN      20   \n",
      "...                                              ...        ...   ...     ...   \n",
      "New cases of confirmed                           ...        NaN   NaN       2   \n",
      "Total new cases registered so far                ...          0     9       5   \n",
      "Total cases of suspects                          ...          0     9       9   \n",
      "Total cases of probables                         ...          2     0      16   \n",
      "Total cases of confirmed                         ...          2     2     378   \n",
      "Cumulative (confirmed + probable + suspects)     ...          4    11     403   \n",
      "Total suspected non-class cases                  ...          2     4     197   \n",
      "New deaths registered                            ...        NaN   NaN      15   \n",
      "Total deaths of suspects                         ...          0     0       0   \n",
      "Total deaths of probables                        ...          2     0      16   \n",
      "Total deaths of confirmed                        ...          2     2     230   \n",
      "Total deaths (confirmed + probables + suspects)  ...          4     0     246   \n",
      "Fatality rate for confirmed and probables        ...       100%  100%     61%   \n",
      "New cases of confirmed among health workers      ...        NaN   NaN     NaN   \n",
      "Total case of confirmed among health workers     ...          0     0       9   \n",
      "New deaths registered among health workers       ...          0     0       0   \n",
      "Total deaths registered among health workers     ...          0     0       7   \n",
      "New admits to CTE so far                         ...        NaN   NaN     NaN   \n",
      "Total number of hospitalized cases in CTE        ...          0   NaN      12   \n",
      "Total number of exits from CTE                   ...          0   NaN      15   \n",
      "Total number of admissions to CTE                ...          0   NaN      27   \n",
      "New contacts registered so far                   ...        NaN   NaN     NaN   \n",
      "Total contacts registered from start date        ...         94     0    1563   \n",
      "Number of contacts to follow today               ...          0     0    1038   \n",
      "Number of contacts followed today                ...          0     0    1038   \n",
      "Number of contacts out of the track 21 days      ...         94     0     525   \n",
      "Number of contacts lost to follow up             ...          0     0       0   \n",
      "Number of samples collected today                ...        NaN   NaN       0   \n",
      "Number of samples under test                     ...        NaN   NaN       0   \n",
      "Total samples tested                             ...        NaN   NaN       0   \n",
      "\n",
      "                                                Mzerekore Nzerekore Pita  \\\n",
      "Description                                                                \n",
      "New cases of suspects                                 NaN         0    0   \n",
      "New cases of probables                                NaN         0    0   \n",
      "New cases of confirmed                                NaN         0    0   \n",
      "Total new cases registered so far                     NaN         0    0   \n",
      "Total cases of suspects                               NaN         0    0   \n",
      "Total cases of probables                              NaN         0    1   \n",
      "Total cases of confirmed                              NaN         4    1   \n",
      "Cumulative (confirmed + probable + suspects)          NaN         4    2   \n",
      "New deaths registered today                           NaN         0    0   \n",
      "New deaths registered today (confirmed)               NaN         0    0   \n",
      "New deaths registered today (probables)               NaN         0    0   \n",
      "New deaths registered today (suspects)                NaN         0    0   \n",
      "Total deaths of suspects                              NaN         0    0   \n",
      "Total deaths of probables                             NaN         0    1   \n",
      "Total deaths of confirmed                             NaN         1    1   \n",
      "Total deaths (confirmed + probables + suspects)       NaN         1    2   \n",
      "Total PEC center today                                NaN         0    0   \n",
      "Total PEC center today (confirmed)                    NaN         0    0   \n",
      "Total PEC center today (probables)                    NaN         0    0   \n",
      "Total PEC center today (suspects)                     NaN         0    0   \n",
      "Total of deaths in confirmed cases in CTE             NaN         0    0   \n",
      "Total of cured in confirmed cases in CTE              NaN         0    0   \n",
      "Number of male confirmed cases                        NaN         2    0   \n",
      "Number of female confirmed cases                      NaN         2    1   \n",
      "Number of male probables cases                        NaN         0    1   \n",
      "Number of female probables cases                      NaN         0    0   \n",
      "Number of male suspects cases                         NaN         0    0   \n",
      "Number of female suspects cases                       NaN         0    0   \n",
      "Total number of male cases                            NaN         2    1   \n",
      "Total number of female cases                          NaN         2    1   \n",
      "...                                                   ...       ...  ...   \n",
      "New cases of confirmed                                NaN       NaN  NaN   \n",
      "Total new cases registered so far                     NaN         1    0   \n",
      "Total cases of suspects                               NaN         2    1   \n",
      "Total cases of probables                              NaN        17    1   \n",
      "Total cases of confirmed                              NaN        31    6   \n",
      "Cumulative (confirmed + probable + suspects)          NaN        50    8   \n",
      "Total suspected non-class cases                       NaN        16   18   \n",
      "New deaths registered                                 NaN       NaN  NaN   \n",
      "Total deaths of suspects                              NaN         0    0   \n",
      "Total deaths of probables                             NaN        17    1   \n",
      "Total deaths of confirmed                             NaN        15    2   \n",
      "Total deaths (confirmed + probables + suspects)       NaN        32    3   \n",
      "Fatality rate for confirmed and probables             NaN       64%  38%   \n",
      "New cases of confirmed among health workers           NaN       NaN  NaN   \n",
      "Total case of confirmed among health workers          NaN         2    0   \n",
      "New deaths registered among health workers            NaN         0    0   \n",
      "Total deaths registered among health workers          NaN         1    0   \n",
      "New admits to CTE so far                              NaN       NaN  NaN   \n",
      "Total number of hospitalized cases in CTE             NaN         0    0   \n",
      "Total number of exits from CTE                        NaN         0    0   \n",
      "Total number of admissions to CTE                     NaN         0    0   \n",
      "New contacts registered so far                        NaN       NaN  NaN   \n",
      "Total contacts registered from start date             NaN       353  140   \n",
      "Number of contacts to follow today                    NaN       126    0   \n",
      "Number of contacts followed today                     NaN       126    0   \n",
      "Number of contacts out of the track 21 days           NaN       227  140   \n",
      "Number of contacts lost to follow up                  NaN         0    0   \n",
      "Number of samples collected today                     NaN       NaN  NaN   \n",
      "Number of samples under test                          NaN       NaN  NaN   \n",
      "Total samples tested                                  NaN       NaN  NaN   \n",
      "\n",
      "                                                Siguiri Telimele Totals Yomou  \n",
      "Description                                                                    \n",
      "New cases of suspects                                 0        0      5   NaN  \n",
      "New cases of probables                                0        0      0   NaN  \n",
      "New cases of confirmed                                0        0      4   NaN  \n",
      "Total new cases registered so far                     0        0      9   NaN  \n",
      "Total cases of suspects                               0        0     11   NaN  \n",
      "Total cases of probables                              0        3    133   NaN  \n",
      "Total cases of confirmed                              6       23    351   NaN  \n",
      "Cumulative (confirmed + probable + suspects)          6       26    495   NaN  \n",
      "New deaths registered today                           0        0      2   NaN  \n",
      "New deaths registered today (confirmed)               0        0      2   NaN  \n",
      "New deaths registered today (probables)               0        0      0   NaN  \n",
      "New deaths registered today (suspects)                0        0      0   NaN  \n",
      "Total deaths of suspects                              0        0      2   NaN  \n",
      "Total deaths of probables                             0        3    133   NaN  \n",
      "Total deaths of confirmed                             3        7    228   NaN  \n",
      "Total deaths (confirmed + probables + suspects)       3       10    363   NaN  \n",
      "Total PEC center today                                2        0     31   NaN  \n",
      "Total PEC center today (confirmed)                    2        0     18   NaN  \n",
      "Total PEC center today (probables)                    0        0      0   NaN  \n",
      "Total PEC center today (suspects)                     0        0     13   NaN  \n",
      "Total of deaths in confirmed cases in CTE             2        5    138   NaN  \n",
      "Total of cured in confirmed cases in CTE              0        9     89   NaN  \n",
      "Number of male confirmed cases                        6        6    161   NaN  \n",
      "Number of female confirmed cases                      0       17    190   NaN  \n",
      "Number of male probables cases                        0        1     52   NaN  \n",
      "Number of female probables cases                      0        2     81   NaN  \n",
      "Number of male suspects cases                         2        0      7   NaN  \n",
      "Number of female suspects cases                       0        0      6   NaN  \n",
      "Total number of male cases                            8        7    220   NaN  \n",
      "Total number of female cases                          0       19    277   NaN  \n",
      "...                                                 ...      ...    ...   ...  \n",
      "New cases of confirmed                              NaN      NaN      6   NaN  \n",
      "Total new cases registered so far                     0        0     34     0  \n",
      "Total cases of suspects                               0        0     45     0  \n",
      "Total cases of probables                              0        3    177     0  \n",
      "Total cases of confirmed                              6       23    977    11  \n",
      "Cumulative (confirmed + probable + suspects)          6       26   1199    11  \n",
      "Total suspected non-class cases                       9       18    974     4  \n",
      "New deaths registered                               NaN      NaN     15   NaN  \n",
      "Total deaths of suspects                              0        0      0     0  \n",
      "Total deaths of probables                             0        3    177     0  \n",
      "Total deaths of confirmed                             3        7    562     5  \n",
      "Total deaths (confirmed + probables + suspects)       3       10    739     5  \n",
      "Fatality rate for confirmed and probables           50%      38%    62%   45%  \n",
      "New cases of confirmed among health workers         NaN      NaN      2   NaN  \n",
      "Total case of confirmed among health workers          2        2     69     1  \n",
      "New deaths registered among health workers            0        0      0     0  \n",
      "Total deaths registered among health workers          0        1     35     1  \n",
      "New admits to CTE so far                            NaN      NaN     25   NaN  \n",
      "Total number of hospitalized cases in CTE             0        0    114     0  \n",
      "Total number of exits from CTE                        0        0     29     0  \n",
      "Total number of admissions to CTE                    13       41   1576     0  \n",
      "New contacts registered so far                      NaN      NaN      0   NaN  \n",
      "Total contacts registered from start date           141      259  11772    56  \n",
      "Number of contacts to follow today                    0        0   2661     0  \n",
      "Number of contacts followed today                     0        0   2595     0  \n",
      "Number of contacts out of the track 21 days         141      259   9110    56  \n",
      "Number of contacts lost to follow up                  0        0      1     0  \n",
      "Number of samples collected today                   NaN      NaN     25   NaN  \n",
      "Number of samples under test                        NaN      NaN      0   NaN  \n",
      "Total samples tested                                NaN      NaN   2832   NaN  \n",
      "\n",
      "[714 rows x 24 columns]\n"
     ]
    }
   ],
   "source": [
    "# INCOMPLETE\n",
    "\n",
    "\n",
    "file_names = glob.glob(DATA + \"/*.csv\")\n",
    "data = pd.DataFrame()\n",
    "data_list = []\n",
    "for file_ in file_names:\n",
    "    df = pd.read_csv(file_,index_col=\"Description\", header=0)\n",
    "    data_list.append(df)\n",
    "data = pd.concat(data_list)\n",
    "\n",
    "print(data)\n",
    "#data = pd.read_csv(DATA + \"/*.csv\")\n",
    "\n",
    "#data = pd.read_csv(\"./Data/ebola/guinea_data/2014-08-04.csv\")\n",
    "#print(data)\n",
    "#[\"Total new cases registered so far\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Task 2. RNA Sequences\n",
    "\n",
    "In the `DATA_FOLDER/microbiome` subdirectory, there are 9 spreadsheets of microbiome data that was acquired from high-throughput RNA sequencing procedures, along with a 10<sup>th</sup> file that describes the content of each. \n",
    "\n",
    "Use pandas to import the first 9 spreadsheets into a single `DataFrame`.\n",
    "Then, add the metadata information from the 10<sup>th</sup> spreadsheet as columns in the combined `DataFrame`.\n",
    "Make sure that the final `DataFrame` has a unique index and all the `NaN` values have been replaced by the tag `unknown`."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "1) Import os (basename) and Glob (in the first cell), which will be used to get the BARCODE and append this information to each row of our dataframe."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "from os.path import basename \n",
    "import os\n",
    "import glob"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "2) Preparing regular expression to fetch all the MID* xls files"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "path = './Data/microbiome'\n",
    "doc_names = glob.glob(path + '/MID*.xls')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "3) Create empty dataframe and data_list which will contain the list of all rows in each file (df) "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "data = pd.DataFrame()\n",
    "data_list = []\n",
    "for file_ in doc_names:\n",
    "    col_name = os.path.splitext(basename(file_))[0] \n",
    "    df = pd.read_excel(file_, index_col=0, names=[\"Value\"], header=None).assign(BARCODE=col_name)\n",
    "    data_list.append(df)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "4) Store all rows from data_list (originally from the 9 files) into our dataframe \"data\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "data = pd.concat(data_list, axis=0).fillna(value = \"unknown\")\n",
    "\n",
    "data.index.name = 'Name'"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "5) Store data from metadata.xls into the \"meta\" dataframe, setting BARCODE as index"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "meta = pd.read_excel(\"./Data/microbiome/metadata.xls\", index_col=\"BARCODE\").fillna(value=\"Na\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "6) Unite information from the \"data\" dataframe with the ones from \"meta\", on the common BARCODE"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "joinresult = data.join(meta, on=\"BARCODE\")\n",
    "joinresult.reset_index(inplace=True)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "7) Setting the new key as Name + BARCODE, this combination is unique"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                                                            Value  \\\n",
      "Name                                               BARCODE          \n",
      "Archaea \"Crenarchaeota\" Thermoprotei Desulfuroc... MID1         7   \n",
      "Archaea \"Crenarchaeota\" Thermoprotei Desulfuroc... MID1         2   \n",
      "Archaea \"Crenarchaeota\" Thermoprotei Sulfolobal... MID1         3   \n",
      "Archaea \"Crenarchaeota\" Thermoprotei Thermoprot... MID1         3   \n",
      "Archaea \"Euryarchaeota\" \"Methanomicrobia\" Metha... MID1         7   \n",
      "Archaea \"Euryarchaeota\" \"Methanomicrobia\" Metha... MID1         1   \n",
      "Archaea \"Euryarchaeota\" \"Methanomicrobia\" Metha... MID1         1   \n",
      "Archaea \"Euryarchaeota\" Archaeoglobi Archaeoglo... MID1         1   \n",
      "Archaea \"Euryarchaeota\" Archaeoglobi Archaeoglo... MID1         1   \n",
      "Archaea \"Euryarchaeota\" Halobacteria Halobacter... MID1         1   \n",
      "Archaea \"Euryarchaeota\" Halobacteria Halobacter... MID1         4   \n",
      "Archaea \"Euryarchaeota\" Halobacteria Halobacter... MID1         1   \n",
      "Archaea \"Euryarchaeota\" Halobacteria Halobacter... MID1         2   \n",
      "Archaea \"Euryarchaeota\" Halobacteria Halobacter... MID1         4   \n",
      "Archaea \"Euryarchaeota\" Methanococci Methanococ... MID1         1   \n",
      "Archaea \"Euryarchaeota\" Methanopyri Methanopyra... MID1        12   \n",
      "Archaea \"Euryarchaeota\" Thermoplasmata Thermopl... MID1         2   \n",
      "Bacteria \"Actinobacteria\" Actinobacteria Actino... MID1         2   \n",
      "Bacteria \"Actinobacteria\" Actinobacteria Actino... MID1         1   \n",
      "Bacteria \"Actinobacteria\" Actinobacteria Actino... MID1         1   \n",
      "Bacteria \"Actinobacteria\" Actinobacteria Actino... MID1         1   \n",
      "Bacteria \"Actinobacteria\" Actinobacteria Actino... MID1         2   \n",
      "Bacteria \"Actinobacteria\" Actinobacteria Actino... MID1        53   \n",
      "Bacteria \"Actinobacteria\" Actinobacteria Actino... MID1         1   \n",
      "Bacteria \"Actinobacteria\" Actinobacteria Actino... MID1         2   \n",
      "Bacteria \"Actinobacteria\" Actinobacteria Actino... MID1         3   \n",
      "Bacteria \"Actinobacteria\" Actinobacteria Actino... MID1         1   \n",
      "Bacteria \"Actinobacteria\" Actinobacteria Actino... MID1        99   \n",
      "Bacteria \"Actinobacteria\" Actinobacteria Actino... MID1        17   \n",
      "Bacteria \"Actinobacteria\" Actinobacteria Actino... MID1         2   \n",
      "...                                                           ...   \n",
      "Bacteria \"Proteobacteria\" Gammaproteobacteria O... MID9        10   \n",
      "Bacteria \"Proteobacteria\" Gammaproteobacteria O... MID9         1   \n",
      "Bacteria \"Proteobacteria\" Gammaproteobacteria O... MID9         1   \n",
      "Bacteria \"Proteobacteria\" Gammaproteobacteria P... MID9         1   \n",
      "Bacteria \"Proteobacteria\" Gammaproteobacteria P... MID9        12   \n",
      "Bacteria \"Proteobacteria\" Gammaproteobacteria P... MID9         4   \n",
      "Bacteria \"Proteobacteria\" Gammaproteobacteria P... MID9       805   \n",
      "Bacteria \"Proteobacteria\" Gammaproteobacteria P... MID9       862   \n",
      "Bacteria \"Proteobacteria\" Gammaproteobacteria P... MID9        12   \n",
      "Bacteria \"Proteobacteria\" Gammaproteobacteria P... MID9      3534   \n",
      "Bacteria \"Proteobacteria\" Gammaproteobacteria T... MID9         4   \n",
      "Bacteria \"Proteobacteria\" Gammaproteobacteria T... MID9         1   \n",
      "Bacteria \"Proteobacteria\" Gammaproteobacteria T... MID9        25   \n",
      "Bacteria \"Proteobacteria\" Gammaproteobacteria T... MID9         2   \n",
      "Bacteria \"Proteobacteria\" Gammaproteobacteria X... MID9         3   \n",
      "Bacteria \"Proteobacteria\" Gammaproteobacteria X... MID9         2   \n",
      "Bacteria \"Proteobacteria\" Gammaproteobacteria X... MID9         1   \n",
      "Bacteria \"Proteobacteria\" Gammaproteobacteria X... MID9         3   \n",
      "Bacteria \"Proteobacteria\" Gammaproteobacteria X... MID9         2   \n",
      "Bacteria \"Proteobacteria\" Gammaproteobacteria X... MID9         1   \n",
      "Bacteria \"Synergistetes\" Synergistia Synergista... MID9         3   \n",
      "Bacteria \"Tenericutes\" Mollicutes Anaeroplasmat... MID9         1   \n",
      "Bacteria \"Tenericutes\" Mollicutes Haloplasmatal... MID9        11   \n",
      "Bacteria \"Thermodesulfobacteria\" Thermodesulfob... MID9         1   \n",
      "Bacteria \"Thermodesulfobacteria\" Thermodesulfob... MID9         1   \n",
      "Bacteria \"Thermotogae\" Thermotogae Thermotogale... MID9        43   \n",
      "Bacteria \"Thermotogae\" Thermotogae Thermotogale... MID9        16   \n",
      "Bacteria \"Verrucomicrobia\" Opitutae Opitutales ... MID9         1   \n",
      "Bacteria Cyanobacteria Cyanobacteria  Chloropla... MID9         1   \n",
      "Bacteria Cyanobacteria Cyanobacteria  Chloropla... MID9        10   \n",
      "\n",
      "                                                                         GROUP  \\\n",
      "Name                                               BARCODE                       \n",
      "Archaea \"Crenarchaeota\" Thermoprotei Desulfuroc... MID1     EXTRACTION CONTROL   \n",
      "Archaea \"Crenarchaeota\" Thermoprotei Desulfuroc... MID1     EXTRACTION CONTROL   \n",
      "Archaea \"Crenarchaeota\" Thermoprotei Sulfolobal... MID1     EXTRACTION CONTROL   \n",
      "Archaea \"Crenarchaeota\" Thermoprotei Thermoprot... MID1     EXTRACTION CONTROL   \n",
      "Archaea \"Euryarchaeota\" \"Methanomicrobia\" Metha... MID1     EXTRACTION CONTROL   \n",
      "Archaea \"Euryarchaeota\" \"Methanomicrobia\" Metha... MID1     EXTRACTION CONTROL   \n",
      "Archaea \"Euryarchaeota\" \"Methanomicrobia\" Metha... MID1     EXTRACTION CONTROL   \n",
      "Archaea \"Euryarchaeota\" Archaeoglobi Archaeoglo... MID1     EXTRACTION CONTROL   \n",
      "Archaea \"Euryarchaeota\" Archaeoglobi Archaeoglo... MID1     EXTRACTION CONTROL   \n",
      "Archaea \"Euryarchaeota\" Halobacteria Halobacter... MID1     EXTRACTION CONTROL   \n",
      "Archaea \"Euryarchaeota\" Halobacteria Halobacter... MID1     EXTRACTION CONTROL   \n",
      "Archaea \"Euryarchaeota\" Halobacteria Halobacter... MID1     EXTRACTION CONTROL   \n",
      "Archaea \"Euryarchaeota\" Halobacteria Halobacter... MID1     EXTRACTION CONTROL   \n",
      "Archaea \"Euryarchaeota\" Halobacteria Halobacter... MID1     EXTRACTION CONTROL   \n",
      "Archaea \"Euryarchaeota\" Methanococci Methanococ... MID1     EXTRACTION CONTROL   \n",
      "Archaea \"Euryarchaeota\" Methanopyri Methanopyra... MID1     EXTRACTION CONTROL   \n",
      "Archaea \"Euryarchaeota\" Thermoplasmata Thermopl... MID1     EXTRACTION CONTROL   \n",
      "Bacteria \"Actinobacteria\" Actinobacteria Actino... MID1     EXTRACTION CONTROL   \n",
      "Bacteria \"Actinobacteria\" Actinobacteria Actino... MID1     EXTRACTION CONTROL   \n",
      "Bacteria \"Actinobacteria\" Actinobacteria Actino... MID1     EXTRACTION CONTROL   \n",
      "Bacteria \"Actinobacteria\" Actinobacteria Actino... MID1     EXTRACTION CONTROL   \n",
      "Bacteria \"Actinobacteria\" Actinobacteria Actino... MID1     EXTRACTION CONTROL   \n",
      "Bacteria \"Actinobacteria\" Actinobacteria Actino... MID1     EXTRACTION CONTROL   \n",
      "Bacteria \"Actinobacteria\" Actinobacteria Actino... MID1     EXTRACTION CONTROL   \n",
      "Bacteria \"Actinobacteria\" Actinobacteria Actino... MID1     EXTRACTION CONTROL   \n",
      "Bacteria \"Actinobacteria\" Actinobacteria Actino... MID1     EXTRACTION CONTROL   \n",
      "Bacteria \"Actinobacteria\" Actinobacteria Actino... MID1     EXTRACTION CONTROL   \n",
      "Bacteria \"Actinobacteria\" Actinobacteria Actino... MID1     EXTRACTION CONTROL   \n",
      "Bacteria \"Actinobacteria\" Actinobacteria Actino... MID1     EXTRACTION CONTROL   \n",
      "Bacteria \"Actinobacteria\" Actinobacteria Actino... MID1     EXTRACTION CONTROL   \n",
      "...                                                                        ...   \n",
      "Bacteria \"Proteobacteria\" Gammaproteobacteria O... MID9              Control 2   \n",
      "Bacteria \"Proteobacteria\" Gammaproteobacteria O... MID9              Control 2   \n",
      "Bacteria \"Proteobacteria\" Gammaproteobacteria O... MID9              Control 2   \n",
      "Bacteria \"Proteobacteria\" Gammaproteobacteria P... MID9              Control 2   \n",
      "Bacteria \"Proteobacteria\" Gammaproteobacteria P... MID9              Control 2   \n",
      "Bacteria \"Proteobacteria\" Gammaproteobacteria P... MID9              Control 2   \n",
      "Bacteria \"Proteobacteria\" Gammaproteobacteria P... MID9              Control 2   \n",
      "Bacteria \"Proteobacteria\" Gammaproteobacteria P... MID9              Control 2   \n",
      "Bacteria \"Proteobacteria\" Gammaproteobacteria P... MID9              Control 2   \n",
      "Bacteria \"Proteobacteria\" Gammaproteobacteria P... MID9              Control 2   \n",
      "Bacteria \"Proteobacteria\" Gammaproteobacteria T... MID9              Control 2   \n",
      "Bacteria \"Proteobacteria\" Gammaproteobacteria T... MID9              Control 2   \n",
      "Bacteria \"Proteobacteria\" Gammaproteobacteria T... MID9              Control 2   \n",
      "Bacteria \"Proteobacteria\" Gammaproteobacteria T... MID9              Control 2   \n",
      "Bacteria \"Proteobacteria\" Gammaproteobacteria X... MID9              Control 2   \n",
      "Bacteria \"Proteobacteria\" Gammaproteobacteria X... MID9              Control 2   \n",
      "Bacteria \"Proteobacteria\" Gammaproteobacteria X... MID9              Control 2   \n",
      "Bacteria \"Proteobacteria\" Gammaproteobacteria X... MID9              Control 2   \n",
      "Bacteria \"Proteobacteria\" Gammaproteobacteria X... MID9              Control 2   \n",
      "Bacteria \"Proteobacteria\" Gammaproteobacteria X... MID9              Control 2   \n",
      "Bacteria \"Synergistetes\" Synergistia Synergista... MID9              Control 2   \n",
      "Bacteria \"Tenericutes\" Mollicutes Anaeroplasmat... MID9              Control 2   \n",
      "Bacteria \"Tenericutes\" Mollicutes Haloplasmatal... MID9              Control 2   \n",
      "Bacteria \"Thermodesulfobacteria\" Thermodesulfob... MID9              Control 2   \n",
      "Bacteria \"Thermodesulfobacteria\" Thermodesulfob... MID9              Control 2   \n",
      "Bacteria \"Thermotogae\" Thermotogae Thermotogale... MID9              Control 2   \n",
      "Bacteria \"Thermotogae\" Thermotogae Thermotogale... MID9              Control 2   \n",
      "Bacteria \"Verrucomicrobia\" Opitutae Opitutales ... MID9              Control 2   \n",
      "Bacteria Cyanobacteria Cyanobacteria  Chloropla... MID9              Control 2   \n",
      "Bacteria Cyanobacteria Cyanobacteria  Chloropla... MID9              Control 2   \n",
      "\n",
      "                                                           SAMPLE  \n",
      "Name                                               BARCODE         \n",
      "Archaea \"Crenarchaeota\" Thermoprotei Desulfuroc... MID1        Na  \n",
      "Archaea \"Crenarchaeota\" Thermoprotei Desulfuroc... MID1        Na  \n",
      "Archaea \"Crenarchaeota\" Thermoprotei Sulfolobal... MID1        Na  \n",
      "Archaea \"Crenarchaeota\" Thermoprotei Thermoprot... MID1        Na  \n",
      "Archaea \"Euryarchaeota\" \"Methanomicrobia\" Metha... MID1        Na  \n",
      "Archaea \"Euryarchaeota\" \"Methanomicrobia\" Metha... MID1        Na  \n",
      "Archaea \"Euryarchaeota\" \"Methanomicrobia\" Metha... MID1        Na  \n",
      "Archaea \"Euryarchaeota\" Archaeoglobi Archaeoglo... MID1        Na  \n",
      "Archaea \"Euryarchaeota\" Archaeoglobi Archaeoglo... MID1        Na  \n",
      "Archaea \"Euryarchaeota\" Halobacteria Halobacter... MID1        Na  \n",
      "Archaea \"Euryarchaeota\" Halobacteria Halobacter... MID1        Na  \n",
      "Archaea \"Euryarchaeota\" Halobacteria Halobacter... MID1        Na  \n",
      "Archaea \"Euryarchaeota\" Halobacteria Halobacter... MID1        Na  \n",
      "Archaea \"Euryarchaeota\" Halobacteria Halobacter... MID1        Na  \n",
      "Archaea \"Euryarchaeota\" Methanococci Methanococ... MID1        Na  \n",
      "Archaea \"Euryarchaeota\" Methanopyri Methanopyra... MID1        Na  \n",
      "Archaea \"Euryarchaeota\" Thermoplasmata Thermopl... MID1        Na  \n",
      "Bacteria \"Actinobacteria\" Actinobacteria Actino... MID1        Na  \n",
      "Bacteria \"Actinobacteria\" Actinobacteria Actino... MID1        Na  \n",
      "Bacteria \"Actinobacteria\" Actinobacteria Actino... MID1        Na  \n",
      "Bacteria \"Actinobacteria\" Actinobacteria Actino... MID1        Na  \n",
      "Bacteria \"Actinobacteria\" Actinobacteria Actino... MID1        Na  \n",
      "Bacteria \"Actinobacteria\" Actinobacteria Actino... MID1        Na  \n",
      "Bacteria \"Actinobacteria\" Actinobacteria Actino... MID1        Na  \n",
      "Bacteria \"Actinobacteria\" Actinobacteria Actino... MID1        Na  \n",
      "Bacteria \"Actinobacteria\" Actinobacteria Actino... MID1        Na  \n",
      "Bacteria \"Actinobacteria\" Actinobacteria Actino... MID1        Na  \n",
      "Bacteria \"Actinobacteria\" Actinobacteria Actino... MID1        Na  \n",
      "Bacteria \"Actinobacteria\" Actinobacteria Actino... MID1        Na  \n",
      "Bacteria \"Actinobacteria\" Actinobacteria Actino... MID1        Na  \n",
      "...                                                           ...  \n",
      "Bacteria \"Proteobacteria\" Gammaproteobacteria O... MID9     stool  \n",
      "Bacteria \"Proteobacteria\" Gammaproteobacteria O... MID9     stool  \n",
      "Bacteria \"Proteobacteria\" Gammaproteobacteria O... MID9     stool  \n",
      "Bacteria \"Proteobacteria\" Gammaproteobacteria P... MID9     stool  \n",
      "Bacteria \"Proteobacteria\" Gammaproteobacteria P... MID9     stool  \n",
      "Bacteria \"Proteobacteria\" Gammaproteobacteria P... MID9     stool  \n",
      "Bacteria \"Proteobacteria\" Gammaproteobacteria P... MID9     stool  \n",
      "Bacteria \"Proteobacteria\" Gammaproteobacteria P... MID9     stool  \n",
      "Bacteria \"Proteobacteria\" Gammaproteobacteria P... MID9     stool  \n",
      "Bacteria \"Proteobacteria\" Gammaproteobacteria P... MID9     stool  \n",
      "Bacteria \"Proteobacteria\" Gammaproteobacteria T... MID9     stool  \n",
      "Bacteria \"Proteobacteria\" Gammaproteobacteria T... MID9     stool  \n",
      "Bacteria \"Proteobacteria\" Gammaproteobacteria T... MID9     stool  \n",
      "Bacteria \"Proteobacteria\" Gammaproteobacteria T... MID9     stool  \n",
      "Bacteria \"Proteobacteria\" Gammaproteobacteria X... MID9     stool  \n",
      "Bacteria \"Proteobacteria\" Gammaproteobacteria X... MID9     stool  \n",
      "Bacteria \"Proteobacteria\" Gammaproteobacteria X... MID9     stool  \n",
      "Bacteria \"Proteobacteria\" Gammaproteobacteria X... MID9     stool  \n",
      "Bacteria \"Proteobacteria\" Gammaproteobacteria X... MID9     stool  \n",
      "Bacteria \"Proteobacteria\" Gammaproteobacteria X... MID9     stool  \n",
      "Bacteria \"Synergistetes\" Synergistia Synergista... MID9     stool  \n",
      "Bacteria \"Tenericutes\" Mollicutes Anaeroplasmat... MID9     stool  \n",
      "Bacteria \"Tenericutes\" Mollicutes Haloplasmatal... MID9     stool  \n",
      "Bacteria \"Thermodesulfobacteria\" Thermodesulfob... MID9     stool  \n",
      "Bacteria \"Thermodesulfobacteria\" Thermodesulfob... MID9     stool  \n",
      "Bacteria \"Thermotogae\" Thermotogae Thermotogale... MID9     stool  \n",
      "Bacteria \"Thermotogae\" Thermotogae Thermotogale... MID9     stool  \n",
      "Bacteria \"Verrucomicrobia\" Opitutae Opitutales ... MID9     stool  \n",
      "Bacteria Cyanobacteria Cyanobacteria  Chloropla... MID9     stool  \n",
      "Bacteria Cyanobacteria Cyanobacteria  Chloropla... MID9     stool  \n",
      "\n",
      "[2396 rows x 3 columns]\n"
     ]
    }
   ],
   "source": [
    "res = joinresult.set_index(['Name', 'BARCODE'])\n",
    "\n",
    "print(res)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "8) Proof of uniqueness of the key"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "True\n"
     ]
    }
   ],
   "source": [
    "print(res.index.is_unique)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Task 3. Class War in Titanic\n",
    "\n",
    "Use pandas to import the data file `Data/titanic.xls`. It contains data on all the passengers that travelled on the Titanic."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "ename": "NameError",
     "evalue": "name 'DATA_FOLDER' is not defined",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mNameError\u001b[0m                                 Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-11-982ede44d58d>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m()\u001b[0m\n\u001b[1;32m      1\u001b[0m \u001b[0;32mfrom\u001b[0m \u001b[0mIPython\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mcore\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mdisplay\u001b[0m \u001b[0;32mimport\u001b[0m \u001b[0mHTML\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m----> 2\u001b[0;31m \u001b[0mHTML\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mfilename\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mDATA_FOLDER\u001b[0m\u001b[0;34m+\u001b[0m\u001b[0;34m'/titanic.html'\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m",
      "\u001b[0;31mNameError\u001b[0m: name 'DATA_FOLDER' is not defined"
     ]
    }
   ],
   "source": [
    "from IPython.core.display import HTML\n",
    "HTML(filename=DATA_FOLDER+'/titanic.html')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "For each of the following questions state clearly your assumptions and discuss your findings:\n",
    "1. Describe the *type* and the *value range* of each attribute. Indicate and transform the attributes that can be `Categorical`. \n",
    "2. Plot histograms for the *travel class*, *embarkation port*, *sex* and *age* attributes. For the latter one, use *discrete decade intervals*. \n",
    "3. Calculate the proportion of passengers by *cabin floor*. Present your results in a *pie chart*.\n",
    "4. For each *travel class*, calculate the proportion of the passengers that survived. Present your results in *pie charts*.\n",
    "5. Calculate the proportion of the passengers that survived by *travel class* and *sex*. Present your results in *a single histogram*.\n",
    "6. Create 2 equally populated *age categories* and calculate survival proportions by *age category*, *travel class* and *sex*. Present your results in a `DataFrame` with unique index."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "# Write your answer here"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.1"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
